### PR TITLE
update(userspace/libsinsp): improve event codes api 

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind cleanup

/kind feature

**Any specific area of the project related to this PR?**

/area libsinsp

**Does this PR require a change in the driver versions?**

No

**What this PR does / why we need it**:

This solves a couple of issues following-up the changes of #877. Moreover, this adds two new functions `libsinsp::events::is_generic` and `libsinsp::events::info`. The latter helps reducing and avoiding the use of the `g_infotables` tables. Moreover, the `libsinsp::events::set` implementation has been improved to have an interface compatible with the standard library, hence making it much easier to use.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: userspace/libscap/examples/01-open/scap_open.c [hunk 1]
```diff
@@ -240,7 +240,7 @@ void print_EF_MODIFIES_STATE_syscalls()
 
 void print_sinsp_modifies_state_syscalls()
 {
-	uint32_t ppm_scs[PPM_SC_MAX];
+	uint8_t ppm_scs[PPM_SC_MAX];
```
Comment (incertum) [inline userspace/libscap/examples/01-open/scap_open.c:243]: Nice!

**Review by incertum — APPROVED** [review]
Merge this PR before https://github.com/falcosecurity/libs/pull/889 and have @FedeDP rebase?

/approve

**Comment by FedeDP** [comment]
Yep! I will rebase mine!

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**